src/main/java/org/truffleruby/core/array/ArrayEnsureCapacityNode.java

```diff
@@ -31,7 +31,28 @@ public static ArrayEnsureCapacityNode create() {
 
     public abstract Object executeEnsureCapacity(DynamicObject array, int requiredCapacity);
 
-    @Specialization(guards = "strategy.matches(array)", limit = "STORAGE_STRATEGIES")
+    @Specialization(guards = { "!strategy.isStorageMutable()", "strategy.matches(array)" }, limit = "ARRAY_STRATEGIES")
```

Comment:
I think we should put guards on separate lines.

Reply:
That's not how it's done in most Array-related nodes so it's probably best to stay consistent.
I usually split on a new line when it becomes too long (>3 guards)